web-ng/src/app/components/map/map.component.ts

```diff
@@ -384,60 +386,113 @@ export class MapComponent implements AfterViewInit, OnDestroy {
+      vertex => new google.maps.LatLng(vertex.latitude, vertex.longitude)
+    );
+
+    this.addPolygonToMap(feature.id, color, layerName, [vertices]);
+  }
+
+  private addPolygonToMap(
+    featureId: string,
+    color: string | undefined,
+    layerName: string | undefined,
+    paths: google.maps.LatLng[][]
+  ) {
     const polygon = new google.maps.Polygon({
-      paths: vertices,
+      paths: paths,
       clickable: true,
       strokeColor: color,
       strokeOpacity: 1,
       strokeWeight: normalPolygonStrokeWeight,
       fillOpacity: 0,
+      map: this.map.googleMap,
     });
+    polygon.set('id', featureId);
+    polygon.set('color', color);
+    polygon.set('layerName', layerName);
+    polygon.addListener('click', (event: google.maps.PolyMouseEvent) => {
+      if (this.disableMapClicks) {
+        return;
+      }
+      const clickedPolygons: google.maps.Polygon[] = [];
+      for (const polygon of this.polygons.values()) {
+        if (google.maps.geometry.poly.containsLocation(event.latLng, polygon)) {
+          clickedPolygons.push(polygon);
+        }
+      }
+      if (clickedPolygons.length === 1) {
+        this.zone.run(() => {
+          this.navigationService.selectFeature(clickedPolygons[0].get('id'));
+        });
+        return;
+      }
 
-    polygon.set('id', feature.id);
-    polygon.addListener('click', () => {
-      polygon.setOptions({ strokeWeight: enlargedPolygonStrokeWeight });
-      this.panAndZoom(vertices[0]);
-      this.navigationService.selectFeature(feature.id);
+      this.zone.run(() => {
+        const dialogRef = this.dialog.open(SelectFeatureDialogComponent, {
+          width: '500px',
+          data: {
+            clickedFeatures: clickedPolygons.map(this.polygonToFeatureData),
+          },
+        });
+        dialogRef.afterClosed().subscribe(result => {
+          if (result) {
+            this.navigationService.selectFeature(result);
+          }
+        });
+      });
     });
-
-    polygon.setMap(this.map.googleMap!);
-
-    this.polygons.set(feature.id, polygon);
+    this.polygons.set(featureId, polygon);
   }
 
-  private updateStylingFunctionForAllGeoJsons(
-    project: Project,
-    selectedFeatureId: string | null
-  ) {
-    this.map.data.setStyle(mapFeature => {
-      const layerId = mapFeature.getProperty('layerId');
-      const color = project.layers.get(layerId)?.color;
-      const featureId = mapFeature.getProperty('featureId');
-      const isSelefted = featureId === selectedFeatureId;
-      return {
-        fillOpacity: 0,
-        strokeColor: color,
-        strokeWeight: isSelefted
-          ? enlargedPolygonStrokeWeight
-          : normalPolygonStrokeWeight,
-      };
-    });
+  private polygonToFeatureData(polygon: google.maps.Polygon): FeatureData {
+    return {
+      featureId: polygon.get('id'),
+      color: polygon.get('color'),
+      layerName: polygon.get('layerName'),
```

Comment:
Why do we need to store the layer name with the feature? If we need this later we should be able to look it up by layerId.

Reply:
Layer name is here for displaying purpose. We could inject layer service and get the layer name by id. But I would rather let the dialog component be a dummy component, and we have everything we need in the map component.